packages/protocol/contracts-0.8/common/MintGoldSchedule.sol

```diff
@@ -0,0 +1,315 @@
+import "../../contracts/common/Initializable.sol";
+import "../../contracts/common/interfaces/IGoldToken.sol";
+
+/**
+ * @title Contract for minting new CELO token based on a schedule.
+ */
+contract MintGoldSchedule is UsingRegistry, ReentrancyGuard, Initializable, IsL2Check {
+  using FixidityLib for FixidityLib.Fraction;
+
+  uint256 constant GENESIS_GOLD_SUPPLY = 600000000 ether; // 600 million Gold
+  uint256 constant GOLD_SUPPLY_CAP = 1000000000 ether; // 1 billion Gold
+  uint256 constant YEARS_LINEAR = 15;
+  uint256 constant SECONDS_LINEAR = YEARS_LINEAR * 365 * 1 days;
+
+  bool public areDependenciesSet;
+  uint256 constant genesisStartTime = 1587587214; // Copied over from `EpochRewards().startTime()`.
+  uint256 public l2StartTime;
+  uint256 public totalSupplyAtL2Start;
+
+  uint256 public totalMintedBySchedule;
+  address public communityRewardFund;
+  address public carbonOffsettingPartner;
+
+  FixidityLib.Fraction private communityRewardFraction;
+  FixidityLib.Fraction private carbonOffsettingFraction;
+
+  event CommunityRewardFractionSet(uint256 fraction);
+  event CarbonOffsettingFundSet(address indexed partner, uint256 fraction);
+
+  modifier whenDependenciesSet() {
+    require(areDependenciesSet, "Minting schedule has not been configured.");
+    _;
+  }
+
+  /**
+   * @notice Sets initialized == true on implementation contracts
+   * @param test Set to true to skip implementation initialization
+   */
+  constructor(bool test) public Initializable(test) {}
+
+  /**
+   * @notice A constructor for initialising a new instance of a MintGoldSchedule contract.
+   */
+  function initialize() external initializer {
+    _transferOwnership(msg.sender);
+  }
+
+  /**
+   * @notice Sets the minting schedule dependencies during L2 transition.
+   * @param _l2StartTime The timestamp of L1 to L2 transition.
+   * @param _communityRewardFraction The percentage of rewards that go the community funds.
+   * @param _carbonOffsettingPartner The address of the carbon offsetting partner.
+   * @param _carbonOffsettingFraction The percentage of rewards going to carbon offsetting partner.
+   * @param registryAddress Address of the deployed contracts registry.
+   */
+  function setDependecies(
+    uint256 _l2StartTime,
+    uint256 _communityRewardFraction,
+    address _carbonOffsettingPartner,
+    uint256 _carbonOffsettingFraction,
+    address registryAddress
+  ) external onlyOwner onlyL2 {
+    require(!areDependenciesSet, "Dependencies have already been set.");
+    require(registryAddress != address(0), "The registry address cannot be the zero address");
+    require(block.timestamp > _l2StartTime, "L2 start time cannot be set to a future date.");
+    areDependenciesSet = true;
+    l2StartTime = _l2StartTime;
+    setRegistry(registryAddress);
+    communityRewardFund = address(getGovernance());
+    totalSupplyAtL2Start = getGoldToken().totalSupply();
+    setCommunityRewardFraction(_communityRewardFraction);
```

Comment:
This will. To avoid importing the `EpochRewards` contract into this one (not included in registry) just to read the fraction variable, the fractions are passed in as a parameter. There should cause minimal confusion, as the function parameters will be defined in the governance proposal that executes this function.

Also opted not to hard code the values, as these values could be changed at anytime before deployment by governance proposal, such as [this](https://celo.stake.id/#/proposal/178) one.